clients/drcachesim/scheduler/scheduler.cpp

```diff
@@ -1943,6 +1954,29 @@ scheduler_tmpl_t<RecordType, ReaderType>::stop_speculation(output_ordinal_t outp
     return sched_type_t::STATUS_OK;
 }
 
+template <typename RecordType, typename ReaderType>
+typename scheduler_tmpl_t<RecordType, ReaderType>::stream_status_t
+scheduler_tmpl_t<RecordType, ReaderType>::set_output_active(output_ordinal_t output,
+                                                            bool active)
+{
+    if (options_.mapping != MAP_TO_ANY_OUTPUT)
+        return sched_type_t::STATUS_INVALID;
+    outputs_[output].active = active;
+    VPRINT(this, 2, "Output stream %d is now %s\n", output,
+           active ? "active" : "inactive");
+    std::lock_guard<std::mutex> guard(sched_lock_);
+    if (!active) {
+        // Make the now-inactive output's input available for other cores.
+        // This will reset its quantum too.
+        // We haven't read ahead so we're "pre_instruction" vs regular switches.
+        inputs_[outputs_[output].cur_input].switching_pre_instruction = true;
```

Comment:
If the output was already inactive, would cur_input be some out-of-bounds index? Maybe return early if `active` is same as the current `outputs_[output].active`.

Reply:
Yes, good catch.  Added the suggestion, and added a test.